cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -1583,6 +1595,14 @@ class ObjectInputFile final : public io::RandomAccessFile {
 // (for rational, see: https://github.com/apache/arrow/issues/34363)
 static constexpr int64_t kPartUploadSize = 10 * 1024 * 1024;
 
+// Above this threshold, use a multi-part upload instead of a single request upload. Only
+// relevant if early sanitization of writing to the bucket is activated.
```

Comment:
Isn't it the reverse? For example:
```suggestion
// Above this threshold, use a multi-part upload instead of a single request upload.
// Only relevant if early sanitization of writing to the bucket is disabled
// (see `allow_delayed_open`).
```